binary/relations.proto

```diff
@@ -88,6 +98,20 @@ message ReadRel {
 
 }
 
+message DistributeRel {
```

Comment:
currently, the DistributeRel is not consistent with substrait doc(https://substrait.io/relations/physical_relations/#distribute-operator). Let's merge as it is for now and change it after rebase the substrait main branch.